detection_rules/kbwrap.py

```diff
@@ -107,59 +116,154 @@ def kibana_import_rules(ctx: click.Context, rules: RuleCollection, overwrite: Op
+
+    # Handle Exceptions Directory Location
+    if results and exceptions_directory:
+        exceptions_directory.mkdir(parents=True, exist_ok=True)
+    exceptions_directory = exceptions_directory or RULES_CONFIG.exception_dir
+    if not exceptions_directory and export_exceptions:
+        click.echo("Warning: Exceptions export requested, but no exceptions directory found")
 
     if results:
         directory.mkdir(parents=True, exist_ok=True)
+    else:
+        click.echo("No rules found to export")
+        return []
+
+    rules_results = results
+    if export_exceptions:
+        # Assign counts to variables
+        rules_count = results[-1]["exported_rules_count"]
+        exception_list_count = results[-1]["exported_exception_list_count"]
+        exception_list_item_count = results[-1]["exported_exception_list_item_count"]
+
+        # Parse rules results and exception results from API return
+        rules_results = results[:rules_count]
+        exception_results = results[rules_count:rules_count + exception_list_count + exception_list_item_count]
 
     errors = []
     exported = []
-    for rule_resource in results:
+    exception_list_rule_table = {}
+    for rule_resource in rules_results:
         try:
-            contents = TOMLRuleContents.from_rule_resource(rule_resource, maturity='production')
-            threat = contents.data.get('threat')
-            first_tactic = threat[0].tactic.name if threat else ''
+            if strip_version:
+                rule_resource.pop("revision", None)
+                rule_resource.pop("version", None)
+
+            contents = TOMLRuleContents.from_rule_resource(rule_resource, maturity="production")
+            threat = contents.data.get("threat")
+            first_tactic = threat[0].tactic.name if threat else ""
             rule_name = rulename_to_filename(contents.data.name, tactic_name=first_tactic)
-            rule = TOMLRule(contents=contents, path=directory / f'{rule_name}')
+            rule = TOMLRule(contents=contents, path=directory / f"{rule_name}")
         except Exception as e:
             if skip_errors:
                 print(f'- skipping {rule_resource.get("name")} - {type(e).__name__}')
                 errors.append(f'- {rule_resource.get("name")} - {e}')
                 continue
             raise
-
+        if rule.contents.data.exceptions_list:
+            # For each item in rule.contents.data.exceptions_list to the exception_list_rule_table under the list_id
+            for exception in rule.contents.data.exceptions_list:
+                exception_id = exception["list_id"]
+                if exception_id not in exception_list_rule_table:
+                    exception_list_rule_table[exception_id] = []
+                exception_list_rule_table[exception_id].append({"id": rule.id, "name": rule.name})
         exported.append(rule)
 
+    # Parse exceptions results from API return
+    exceptions = []
+    if export_exceptions:
```

Comment:
nit: this cli command is getting a bit long. May be worth refactoring into a method starting with this exceptions loading code below.

Reply:
I agree, will take a look once https://github.com/elastic/detection-rules/pull/3955 is merged.

Reply:
Done